File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.c


int maxProfit(int* prices, int pricesSize){
    int i,min = prices[0], profit = 0;
    for(i=1;i<pricesSize;i++){
        if(prices[i]<min)
            min = prices[i];
        else if(prices[i]-min>profit)
            profit = prices[i]-min;
    }
    return profit;
}

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        n,mini=len(prices),prices[0]
        ans=0
        for i in range(1,n):
            if prices[i]>mini:
                ans=max(ans,prices[i]-mini)
            else:
                mini=prices[i]
        return ans